query/src/interpreters/interpreter_copy.rs

```diff
@@ -69,9 +76,55 @@ impl Interpreter for CopyInterpreter {
         &self,
         mut _input_stream: Option<SendableDataBlockStream>,
     ) -> Result<SendableDataBlockStream> {
-        let stage_info = self.plan.stage_plan.stage_info.clone();
-        match stage_info.stage_params.storage {
-            StageStorage::S3(s3) => {}
-        }
+        let dal = self
+            .get_dal(self.plan.stage_plan.stage_info.clone())
+            .await?;
+
+        let max_block_size = self.ctx.get_settings().get_max_block_size()? as usize;
+        let path = self
+            .plan
+            .stage_plan
+            .stage_info
+            .stage_params
+            .location
+            .clone()
+            .as_str();
+        let o = dal.stat(path).run().await.unwrap();
+        let reader = SeekableReader::new(dal, path, o.size);
+        let read_buffer_size = self.ctx.get_settings().get_storage_read_buffer_size()?;
+        let reader = BufReader::with_capacity(read_buffer_size as usize, reader);
+
+        let source_params = SourceParams {
+            reader,
+            path,
+            format: self.plan.format.as_str(),
+            schema: self.plan.schema.clone(),
+            max_block_size,
+            projection: (0..self.plan.schema().fields().len()).collect(),
+            options: &self.plan.options,
+        };
+        let source_stream = SourceStream::new(SourceFactory::try_get(source_params)?);
+        let input_stream = source_stream.execute().await?;
+        let progress_stream = Box::pin(ProgressStream::try_create(
+            input_stream,
+            self.ctx.get_scan_progress(),
+        )?);
+
+        let table = self
+            .ctx
+            .get_table(&self.plan.db_name, &self.plan.tbl_name)
+            .await?;
```

Comment:
Maybe we can verify `db_name` and `tbl_name` before accessing file in s3? If wrong db_name/tbl_name is given we can fail.

Reply:
By the way, I'm fresh on this project -- databend, if some naive comments posted, I'm supposed to offend you guys. Be free to give correct replies.

Reply:
This PR is under Draft status, `databse` and `table` privileges check hasn't finished yet, will add them soon.
Thank you.

Reply:
Do you have some recent deadline about this feature? If not, I'd like to write some code for it.

Reply:
For `COPY INTO TABLE FROM s3 external location` feature, I will finish it this week in this PR :)
Well, would like to recommend some issues here:
https://github.com/datafuselabs/databend/issues?q=is%3Aissue+is%3Aopen+label%3A%22C-good+first+issue%22